pirateup/store/controllers.py

```diff
@@ -14,6 +15,11 @@ def stores():
     result = {}
     if request.method == 'POST':
```

Comment:
I think there should be a more specific handler for each action to separate low level code from business logic => eg. on POST you want to:
1. get shop data
2. validate them
3. if they validate => save them to db

So the code should look more like:
```python
if request.method == 'POST':
    shopData = getRequestPayload()
    if validate (shopData):
        addShopToDb(shopData)
    ...
```

And each of the high-level function should have the specific code that you have written (extracting location, etc).

Reply:
damn, deep shit ;)